Fix Notice component action button margins

Infinite-Null opened the pull request:
Closes: #69426

## What?
Improves spacing in Notice component between content and action buttons.

## Why?
The spacing between the main content and action buttons in the Notice component was too tight, and there was no vertical spacing between multiple notice actions.

## Screenshots:

|Before|After|
|-|-|
|![image](https://github.com/user-attachments/assets/f47a273e-e095-4908-8e55-32f48d7fdca1)|![image](https://github.com/user-attachments/assets/bc9dfaf5-2d3d-48c9-afb0-42dc7e6be7d0)|

Infinite-Null committed:
Fix Notice component action button margins

Infinite-Null commented:
Hi @daledupreez,
I have implemented a fix for the notice component spacing please test it and review it at your convenience.
Thank you.

Infinite-Null committed:
Add entry for fix notice component spacing issue

aduth commented on packages/components/src/notice/style.scss:
```diff
@@ -36,20 +36,17 @@
 	}
 
 	// When it has better support, this can be replaced
 	// with column-gap since these are flex items.
```
We can remove this comment now that we're using `gap`. 


```suggestion
```

aduth commented on packages/components/src/notice/style.scss:
```diff
@@ -36,20 +36,17 @@
 	// When it has better support, this can be replaced
 	// with column-gap since these are flex items.
-	margin-right: $grid-unit-10;
+	margin-top: $grid-unit-10;
```
Part of me feels it'd be better to have the margin apply to the container of the actions, although based on the current implementation [we're aways rendering the actions container even if there's no actions](https://github.com/WordPress/gutenberg/blob/a524774b6bc4384045859698a66b9bef8f5fa74a/packages/components/src/notice/index.tsx#L114), and we wouldn't want the extra margin if there's no actions.

In my opinion we probably shouldn't be rendering that container if there's no actions and then it'd be fine to apply the styling, but that's a bit more extensive change.

aduth replied:
Referencing [the Figma designs for this component](https://www.figma.com/design/804HN2REV2iap2ytjRQ055/WordPress-Design-System?node-id=16432-131127&t=yjqzIJbdM70RINPy-0), I think this margin should be `$grid-unit-15` (12px) instead of `$grid-unit-10` (8px), consistent with the gap between each action button.


```suggestion
	margin-top: $grid-unit-15;
```

aduth reviewed (commented):
Thanks for submitting this fix, and apologies for the delay in review. This feels like the right fix for a pretty obviously-broken appearance of actions. I'm going to do a bit more testing before approving, but in general the changes are looking good.

aduth committed:
Remove outdated comment

aduth committed:
Increase margin between actions and content

Align to Figma designs

aduth committed:
Avoid rendering notice actions container unless actions exist

Apply margin styling on container, while avoiding margin affecting layout if there are no actions

aduth committed:
Clarify CHANGELOG entry as fixing actions specifically

aduth reviewed (approved):
I pushed a couple changes to your branch from the feedback I had, but I think this is looking great 👍  

@Infinite-Null How do you feel about the additional changes I included, notably around only rendering the actions container when there are actions to be shown?

t-hamano reviewed (commented):
This PR works fine for the `Notice` component alone, but in the `NoticeList` component, the spacing is larger than intended.

For example, open a post editor and run the following code in your browser console:

<details><summary>Details</summary>

```javascript
wp.data.dispatch( 'core/notices' ).createNotice( 'warning', 'Post published.', {
	isDismissible: true,
	actions: [
		{
			label: 'Click me!',
			variant: 'primary',
		},
		{
			label: 'Or click me instead!',
		},
		{
			label: 'Or visit a link for more info',
			url: 'https://wordpress.org',
			variant: 'link',
		},
		{
			label: 'Click me!',
			variant: 'primary',
		},
		{
			label: 'Or click me instead!',
		},
		{
			label: 'Or visit a link for more info',
			url: 'https://wordpress.org',
			variant: 'link',
		},
	],
} );
```

</details> 

### trunk

<img width="738" height="500" alt="Image" src="https://github.com/user-attachments/assets/e794f662-86ea-4df9-9235-daeedac7d091" />

### This PR

<img width="737" height="538" alt="Image" src="https://github.com/user-attachments/assets/9196168c-d5a1-44f4-a231-65f4e9cc18db" />

I think we can now remove the style here as well:

https://github.com/WordPress/gutenberg/blob/48a6e8ae52d97441b0bbb593162124eb6c74dd46/packages/components/src/notice/style.scss#L90-L94

<img width="735" height="521" alt="Image" src="https://github.com/user-attachments/assets/2168e493-fe5f-4966-988b-866d58269c75" />

aduth committed:
NoticesList: Update story to include variation with actions

aduth committed:
NoticeList: Fix excess vertical spacing on actions

aduth committed:
Notices: Vertically center action buttons

aduth committed:
Notice: Remove unnecessary styling around action buttons

t-hamano reviewed (approved):
LGTM! It looks good on the post editor as well.

| Before | After |
|--------|--------|
| <img width="785" height="508" alt="Image" src="https://github.com/user-attachments/assets/dd2959df-d6c1-43fc-8dfd-146ec9763a4d" /> | <img width="783" height="488" alt="Image" src="https://github.com/user-attachments/assets/fbdf6a91-0d8c-4e6a-9e02-20d2fbcff56b" /> |

aduth commented:
Thanks @Infinite-Null for the fix, and @t-hamano for the review 🙌
